pages/add-clinic.vue

```diff
@@ -124,29 +124,61 @@ export default {
     prefectureList: json.prefectures,
     prefecture: "",
     city: "",
-    cityRules: [
-      (v) => !!v || "City is required",
-      (v) => (v && v.length >= 2) || "City name must be at least 2 characters",
-    ],
     ward: "",
-    wardRules: [
-      (v) => !!v || "Ward is required",
-      (v) => (v && v.length >= 2) || "Ward name must be at least 2 characters",
-    ],
     name: "",
-    nameRules: [
-      (v) => !!v || "Clinic name is required",
-      (v) =>
-        (v && v.length >= 2) || "Clinic name must be at least 2 characters",
-    ],
     note: "",
     website: "",
-    websiteRules: [
-      (v) => !!v || "Website URL is required",
-      (v) => (v && v.length >= 5) || "Please enter a valid URL",
-    ],
   }),
 
+  computed: {
+    cityRules() {
+      const minLength = 2;
+
+      return [
+        (v) => !!v || this.$i18n.t("add-clinic.validations.cityRequired"),
+        (v) =>
+          (v && v.length >= minLength) ||
+          this.$i18n.t("add-clinic.validations.cityValidation", [minLength]),
+      ];
+    },
```

Comment:
Very good! I think it is even more readable now, and the `2` isn't a "magic number" anymore: it states its meaning, a "min length"

Reply:
Thanks! But this part conflicts with #169 but maybe I'll have to close this PR in favour of Vuelidate .